src/main/java/uk/gov/hmcts/befta/dse/ccd/DefinitionConverter.java

```diff
@@ -15,27 +15,31 @@
 
     /**
      * Main method to convert between json and excel versions of a definition file
+     * 
      * @param args
-     *  arg1: to-json | to-excel : key word to convert from excel to json or from json to excel
-     *  arg2 input file path for excel document or parent jurisdiction folder for json version
-     *  arg3 (Optional) output folder path for resulting json or excel file. By default will use parent folder from the input location
-     *  arg4 (Optional) Boolean: true - use jurisdiction name to generate the parent folder name when converting from excel to JSON,
-     *      false - use file name as the folder name
+     * 
+     * arg1: to-json | to-excel : key word to convert from excel to json or from json to excel
+     * arg2 input file path for excel document or parent jurisdiction folder for json version
+     * arg3 (Optional) target environment for which an Excel file is to be generated (with to-excel)
+     * arg4 (Optional) output folder path for resulting json or excel file. By default will use parent folder from the input location 
+     * arg5 (Optional) Boolean: true - use jurisdiction  name to generate the parent folder name when converting from excel to JSON, false - use file name as the folder name
      */
     public static void main(String[] args) {
         validateArgs(args);
 
         String transformationKeyword = args[0];
         String inputPath = args[1];
-        String outputPath = args.length > 2 ? args[2] : null;
-        boolean useJurisdictionAsFolder = args.length <= 3 || Boolean.parseBoolean(args[3]);
+        String forEnvironment = args[2];
+        String outputPath = args.length > 3 ? args[3] : null;
+        boolean useJurisdictionAsFolder = args.length <= 4 || Boolean.parseBoolean(args[4]);
```

Comment:
This arg mapping does not work for the **ccd-test-definitions** `definitionsToJson` command as it executes [`"to-json"` conversion with 4 args](https://github.com/hmcts/ccd-test-definitions/blob/master/build.gradle#L103-L107).  (NB: tested against 8.0.7: which generated error _"Third arg should be a path not a boolean.."_).

.. so either the new optional argument ('environment') **that only applies to `"to-excel"`** should be the last argument or the mappings should be adjusted so they apply differently to `"to-excel"` and `"to-json"` formats.

Reply:
Corrected.